procedures.sql

```diff
@@ -0,0 +1,1136 @@
+            VALUES (@basket_id, @webinar_id);
+        END
+END;
+GO
+-- EXEC AddItemToBasket 17, @course_id = 3
+
+CREATE PROCEDURE RemoveItemFromBasket
+    @basket_id INT,
+    @course_id INT = NULL,
+    @meeting_id INT = NULL,
+    @studies_id INT = NULL,
+    @webinar_id INT = NULL
+AS
+BEGIN
+    IF @course_id IS NOT NULL
+        BEGIN
+            DELETE FROM BasketItem
+            WHERE basket_id = @basket_id AND course_id = @course_id;
+        END
+
+    IF @meeting_id IS NOT NULL
+        BEGIN
+            DELETE FROM BasketItem
+            WHERE basket_id = @basket_id AND meeting_id = @meeting_id;
+        END
+
+    IF @studies_id IS NOT NULL
+        BEGIN
+            DELETE FROM BasketItem
+            WHERE basket_id = @basket_id AND studies_id = @studies_id;
+        END
+
+    IF @webinar_id IS NOT NULL
+        BEGIN
+            DELETE FROM BasketItem
+            WHERE basket_id = @basket_id AND webinar_id = @webinar_id;
+        END
+END;
+GO
+-- EXEC RemoveItemFromBasket 35, @course_id = 3
+
+CREATE PROCEDURE FailBasketPayment
+@basket_id INT
+AS
+BEGIN
+    UPDATE Basket
+    SET state = 'failed_payment'
+    WHERE id = @basket_id;
+END;
+GO
+-- EXEC FailBasketPayment 35
+
+CREATE PROCEDURE ApproveBasketPayment
+@basket_id INT
+AS
+BEGIN
+    DECLARE @student_id INT = (SELECT student_id FROM Basket WHERE id = @basket_id)
+    DECLARE @current_course INT
+    DECLARE @current_meeting INT
+    DECLARE @current_studies INT
+    DECLARE @current_webinar INT
+
+    DECLARE BasketItemCursor CURSOR FOR
+        SELECT course_id, meeting_id, studies_id, webinar_id
+        FROM BasketItem
+        WHERE basket_id = @basket_id
+
+    OPEN BasketItemCursor
+    FETCH NEXT FROM BasketItemCursor INTO @current_course, @current_meeting, @current_studies, @current_webinar
+
+    WHILE @@FETCH_STATUS = 0
+        BEGIN
+            IF @current_course IS NOT NULL
+                BEGIN
+                    EXEC AddStudentToCourse @student_id, @current_course, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP;
+                END
+            IF @current_meeting IS NOT NULL
+                BEGIN
+                    EXEC AddStudentToMeeting @student_id, @current_meeting, CURRENT_TIMESTAMP;
+                END
+            IF @current_studies IS NOT NULL
+                BEGIN
+                    EXEC AddStudentToStudies @student_id, @current_studies, CURRENT_TIMESTAMP;
+                END
+            IF @current_webinar IS NOT NULL
+                BEGIN
+                    EXEC AddStudentToWebinar @student_id, @current_webinar, CURRENT_TIMESTAMP;
+                END
+
+            FETCH NEXT FROM BasketItemCursor INTO @current_course, @current_meeting, @current_studies, @current_webinar
+        END;
+
+    CLOSE BasketItemCursor
+    DEALLOCATE BasketItemCursor
+
+    UPDATE Basket
+    SET state = 'success_payment'
+    WHERE id = @basket_id;
+END;
+GO
+-- EXEC ApproveBasketPayment 35
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
```

Comment:
fill with
```sql
-- A sonnet in the language of the code,
-- Where tables dance and relationships bode.

-- First, select the verses from the dataset,
-- Where rows and columns intertwine, be set.

-- Inner join, the lines elegantly blend,
-- Each word connected, relationships transcend.

-- Where clauses whisper conditions wise,
-- Filtering stanzas, where the meaning lies.

-- Order by, the lines arrange their fate,
-- Ascending or descending, a rhythmic state.

-- Group by, a chorus, a harmonious song,
-- Aggregates sing, as data dance along.

-- A comment here, a note to self,
-- A code poet's tale, on the database shelf.

-- Sixteen lines, a poem complete,
-- In SQL, where data and verses meet.
```

Reply:
I forgot to remove these blind lines and was about to do it now, but your idea is much much more ingenious

Reply:
🤡